### PR TITLE
Fluent: suggested actions roving focus

### PR DESCRIPTION
**Pull request by OEvgeny**
> Fixes #

## Changelog Entry

- (Experimental) Added `botframework-webchat-fluent-theme` package for applying Fluent UI theme to Web Chat, by [@compulim](https://github.com/compulim) and [@OEvgeny](https://github.com/OEvgeny)
   - Added keyboard navigation support into suggested actions, by [@OEvgeny](https://github.com/OEvgeny) in PR [#5154](https://github.com/microsoft/BotFramework-WebChat/pull/5154)

## Description

Implemented roving focus using existing implementation as a reference. The new component is now used for suggested actions keyboard support.

## Design

The new `rovingFocus` module implements the roving focus exposing only the `RovingFocusProvider` and `useRovingFocusItemRef`.

The module is private for `suggestedActions` for now. When we find other use-case, we can reconsider and refactor it.

## Specific Changes

- Implemented the roving focus functionality for fluent theme package
- Added roving focus into suggested actions
- Added tests

## -

-

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [ ] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [ ] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/fluent-theme/src/components/suggestedActions/private/rovingFocus.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    [updateItemTabIndex, itemRefsRef, activeItemIndexRef]
+  );
+
+  const handleFocus = useCallback(
+    event => {
+      const { target } = event;
+
+      const index = itemRefsRef.current.findIndex(({ current }) => current === target);
+
+      // prevent focus the last element, if not found the focused element
```
Comment (OEvgeny) [inline packages/fluent-theme/src/components/suggestedActions/private/rovingFocus.tsx:75]:
```suggestion
      // prevent focusing the last element, if we didn't found the element focused
```

File: packages/fluent-theme/src/components/suggestedActions/SuggestedAction.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import { hooks } from 'botframework-webchat-component';
 import { type DirectLineCardAction } from 'botframework-webchat-core';
 import cx from 'classnames';
-import React, { MouseEventHandler, memo, useCallback, useRef } from 'react';
+import React, { MouseEventHandler, memo, useCallback } from 'react';
```
Comment (compulim) [inline packages/fluent-theme/src/components/suggestedActions/SuggestedAction.tsx:4]: Not your fault. No need to fix for now. We will have a huge push to clean up imports.

```suggestion
import React, { type MouseEventHandler, memo, useCallback } from 'react';
```

File: packages/fluent-theme/src/components/suggestedActions/SuggestedAction.tsx [hunk 3]
```diff
@@ -38,13 +39,14 @@ function SuggestedAction({
   imageAlt,
   text,
   type,
-  value
+  value,
+  itemIndex
```
Comment (compulim) [inline packages/fluent-theme/src/components/suggestedActions/SuggestedAction.tsx:43]: Sort.

File: packages/fluent-theme/src/components/suggestedActions/SuggestedAction.tsx (hunk 2 above)
Comment (OEvgeny) [reply packages/fluent-theme/src/components/suggestedActions/SuggestedAction.tsx:4]: Mine PR on code hygiene has a lint rule enforcing this.

**Review by compulim — APPROVED** [review]

**Pull request merged**